lib/grpcServer/handlers/platform/getDataContractHandlerFactory.js

```diff
@@ -38,31 +37,18 @@ function getDataContractHandlerFactory(driveClient, handleAbciResponseError) {
       throw new InvalidArgumentGrpcError('id is not specified');
     }
 
-    let dataContractBuffer;
-    let proofObject;
+    let dataContractResponseBuffer;
     try {
-      ({ data: dataContractBuffer, proof: proofObject } = await driveClient
-        .fetchDataContract(id, prove));
+      dataContractResponseBuffer = await driveClient
+        .fetchDataContract(id, prove);
     } catch (e) {
       if (e instanceof AbciResponseError) {
         handleAbciResponseError(e);
       }
       throw e;
     }
 
-    const response = new GetDataContractResponse();
-
-    response.setDataContract(dataContractBuffer);
-
-    if (prove === true) {
-      const proof = new Proof();
-      proof.setRootTreeProof(proofObject.rootTreeProof);
-      proof.setStoreTreeProof(proofObject.storeTreeProof);
-
-      response.setProof(proof);
-    }
-
-    return response;
+    return GetDataContractResponse.deserializeBinary(dataContractResponseBuffer);
```

Comment:
Did you try to just return a buffer?

Reply:
Um, I think it'll not work since we do our little conversions using gRPC classes aren't we?

Reply:
Ah, right. Ok, let’s keep it like this.